src/app/src/main/java/com/example/myapplication/WelcomeActivity.java

```diff
@@ -28,20 +36,46 @@ protected void onCreate(Bundle savedInstanceState) {
             v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
             return insets;
         });
-        btn=findViewById(R.id.welcome_button);
-        btn.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
 
+        btn = findViewById(R.id.welcome_button);
+        btn.setVisibility(View.GONE);
 
-                Intent i = new Intent(WelcomeActivity.this, SignUpActivity.class);
-                startActivity(i);
-                finish();
+        String deviceId = Settings.Secure.getString(getContentResolver(), Settings.Secure.ANDROID_ID);
 
+        FirebaseFirestore db = FirebaseFirestore.getInstance();
 
-            }
+        db.collection("users").document(deviceId).get()
```

Comment:
please use the DatabaseManager instead as it takes care of everything for you and builds everything connected to the user

Reply:
i'll update it to reflect your requested changes, it was just redundant as we didn't need to build a user object here, just merely flag whether it was present in the database or not